rtos/rtx/TARGET_CORTEX_M/RTX_CM_lib.h

```diff
@@ -350,7 +350,13 @@ __attribute__((used)) void _mutex_release (OS_ID *mutex) {
 
 /* Main Thread definition */
 extern void pre_main (void);
-osThreadDef_t os_thread_def_main = {(os_pthread)pre_main, osPriorityNormal, 1U, 0U, NULL};
+
+#if defined(TARGET_MCU_NRF51822) || defined(TARGET_MCU_NRF52832)
+static uint32_t thread_stack_main[DEFAULT_STACK_SIZE / sizeof(uint32_t)];
+#else
+static uint32_t thread_stack_main[DEFAULT_STACK_SIZE * 2 / sizeof(uint32_t)];
```

Comment:
The stack should also be 8 bytes aligned, it is a requirement for IAR: https://www.iar.com/support/tech-notes/general/problems-with-printf-floating-point-f-on-arm/
Note: maybe it is done by the OS at a latter stage.